### DISCUSSION TITLE
[Share] Toggle Side Panes (Library/Reader)

### DISCUSSION THREAD

**Discussion by windingwind** [windingwind/zotero-actions-tags#169] [Action Scripts]
### Description

Toggle the left and right panes in the library, as well as in PDF/EPUB/HTML reader tab.

![](https://user-images.githubusercontent.com/33902321/268162587-e9ce0622-798f-447f-9e7c-7bfcb7563ab2.png)

See https://github.com/windingwind/zotero-actions-tags/discussions/118 and https://github.com/windingwind/zotero-actions-tags/discussions/166

### Event

None

### Operation

Script

#### Toggle Left Pane

```js
/**
 * Toggle Left Pane
 * @author [MuiseDestiny](https://github.com/MuiseDestiny), windingwind
 * @usage Assign a shortcut
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/169
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/169
 */
const Zotero = require("Zotero");
const Zotero_Tabs = require("Zotero_Tabs");
const document = require("document");

if (item) {
  return;
}

if (Zotero_Tabs.selectedType === "library") {
  const splitter = document.querySelector("#zotero-collections-splitter");
  if (splitter.getAttribute("state") == "collapsed") {
    splitter.setAttribute("state", "");
    return;
  } else {
    splitter.setAttribute("state", "collapsed");
    return;
  }
} else {
  Zotero.Reader.getByTabID(
    Zotero_Tabs.selectedID
  )._internalReader.toggleSidebar();
  return;
}
```

#### Toggle Right Pane

```js
/**
 * Toggle Right Pane
 * @author [MuiseDestiny](https://github.com/MuiseDestiny), windingwind
 * @usage Assign a shortcut
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/169
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/169
 */
const Zotero = require("Zotero");
const Zotero_Tabs = require("Zotero_Tabs");
const document = require("document");
const CustomEvent = require("CustomEvent");

// Modify the line below for opening item/notes pane by default when in the reader
const DEFAULT_OPEN_PANE = "item";
// const DEFAULT_OPEN_PANE = "notes";

if (item) {
  return;
}

if (Zotero_Tabs.selectedType === "library") {
  const splitter = document.querySelector("#zotero-items-splitter");
  if (splitter.getAttribute("state") == "collapsed") {
    splitter.setAttribute("state", "");
    return;
  } else {
    splitter.setAttribute("state", "collapsed");
    return;
  }
} else {
  document.getElementById(`zotero-tb-toggle-${DEFAULT_OPEN_PANE}-pane`).dispatchEvent(new CustomEvent("click"))
  return;
}
```

**Comment by xiongfuai** [comment]
右侧窗格切换还是有问题！！！

**Comment by sjhsbhqf** [comment]
> ### Description
> Toggle the left and right panes in the library, as well as in PDF/EPUB/HTML reader tab.
> 
> ![](https://user-images.githubusercontent.com/33902321/268162587-e9ce0622-798f-447f-9e7c-7bfcb7563ab2.png)
> 
> See #118 and #166
> 
> ### Event
> None
> 
> ### Operation
> Script
> 
> ### Data
> #### Toggle Left Pane
> ```js
> /**
>  * Toggle Left Pane
>  * @author [MuiseDestiny](https://github.com/MuiseDestiny), windingwind
>  * @usage Assign a shortcut
>  * @link https://github.com/windingwind/zotero-actions-tags/discussions/169
>  * @see https://github.com/windingwind/zotero-actions-tags/discussions/169
>  */
> const Zotero = require("Zotero");
> const Zotero_Tabs = require("Zotero_Tabs");
> const document = require("document");
> 
> if (item) {
>   return;
> }
> 
> if (Zotero_Tabs.selectedType === "library") {
>   const splitter = document.querySelector("#zotero-collections-splitter");
>   if (splitter.getAttribute("state") == "collapsed") {
>     splitter.setAttribute("state", "");
>     return;
>   } else {
>     splitter.setAttribute("state", "collapsed");
>     return;
>   }
> } else {
>   Zotero.Reader.getByTabID(
>     Zotero_Tabs.selectedID
>   )._internalReader.toggleSidebar();
>   return;
> }
> ```
> 
> #### Toggle Right Pane
> ```js
> /**
>  * Toggle Right Pane
>  * @author [MuiseDestiny](https://github.com/MuiseDestiny), windingwind
>  * @usage Assign a shortcut
>  * @link https://github.com/windingwind/zotero-actions-tags/discussions/169
>  * @see https://github.com/windingwind/zotero-actions-tags/discussions/169
>  */
> const Zotero = require("Zotero");
> const Zotero_Tabs = require("Zotero_Tabs");
> const document = require("document");
> const CustomEvent = require("CustomEvent");
> 
> // Modify the line below for opening item/notes pane by default when in the reader
> const DEFAULT_OPEN_PANE = "item";
> // const DEFAULT_OPEN_PANE = "notes";
> 
> if (item) {
>   return;
> }
> 
> if (Zotero_Tabs.selectedType === "library") {
>   const splitter = document.querySelector("#zotero-items-splitter");
>   if (splitter.getAttribute("state") == "collapsed") {
>     splitter.setAttribute("state", "");
>     return;
>   } else {
>     splitter.setAttribute("state", "collapsed");
>     return;
>   }
> } else {
>   document.getElementById(`zotero-tb-toggle-${DEFAULT_OPEN_PANE}-pane`).dispatchEvent(new CustomEvent("click"))
>   return;
> }
> ```

最新的beta56版本这个不能用了

**Reply by Adammartain** [reply]
求更新适配新版本

**Comment by MuiseDestiny** [comment]
右侧，支持主界面+阅读状态下使用
```js
/**
 * Toggle Right Pane
 * @author [MuiseDestiny](https://github.com/MuiseDestiny), windingwind
 * @usage Assign a shortcut
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/169
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/169
 */
const Zotero = require("Zotero");
const Zotero_Tabs = require("Zotero_Tabs");
const document = require("document");
const DEFAULT_OPEN_PANE = "item";

if (item) {
  return;
}

const splitter = document.querySelector(Zotero_Tabs.selectedType === "library" ? "#zotero-items-splitter":"#zotero-context-splitter");

if (splitter.getAttribute("state") == "collapsed") {
splitter.setAttribute("state", "");
return;
} else {
splitter.setAttribute("state", "collapsed");
return;
}

```

**Reply by Adammartain** [reply]
谢谢大佬！真快哈哈哈

**Reply by MickeyMiao** [reply]
求更新左侧窗格

**Reply by e-kotov** [reply]
works beautifully, thanks!

**Reply by LHZXY0** [reply]
item is not definited 该怎么办

**Comment by mjthoraval** [comment]
Thank you very much for the update.
If I want to combine the two actions in one, is there a simple way to do it? So that updating one of the scripts will automatically update the combined one?
I see that there is an operation "Trigger Another Action". Can I use that somehow to make a new action for "Toggle Side Panes"?

**Reply by FeralFlora** [reply]
"Trigger Another Action" I can't seem to make this work either. I think this operation is undocumented, @windingwind?

I just set the same hotkey to both scripts to be able to toggle both sidebars at the same time. But of course, this means that I lost the ability to toggle only one of them. Also, since it toggles, rather than expand, it can mean that one side closes while the other opens.

**Reply by windingwind** [reply]
Thanks for reminding me. I updated the readme: https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#action-settings and released a new version that supports triggering multiple actions.

**Reply by FeralFlora** [reply]
> Thanks for reminding me. I updated the readme: https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#action-settings and released a new version that supports triggering multiple actions.

You're welcome! I tried adding the two actions like so, both it's not working (nothing happens):

![image](https://github.com/windingwind/zotero-actions-tags/assets/33608083/1460fd04-c5aa-4497-9c17-b403c4dc52f2)


I tried with and without quotes, but neither way works.

**Reply by windingwind** [reply]
You need to update to the latest beta version of the plugin first (and probably with Zotero restart)

**Reply by ramonmi** [reply]
> You need to update to the latest beta version of the plugin first (and probably with Zotero restart)

Still doesn't work.

**Reply by windingwind** [reply]
If it still does not work, please report in a new issue.

**Reply by ramonmi** [reply]
> > Thanks for reminding me. I updated the readme: https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#action-settings and released a new version that supports triggering multiple actions.
> 
> You're welcome! I tried adding the two actions like so, both it's not working (nothing happens):
> 
> ![image](https://private-user-images.githubusercontent.com/33608083/337299419-1460fd04-c5aa-4497-9c17-b403c4dc52f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc5NDM0MjUsIm5iZiI6MTcxNzk0MzEyNSwicGF0aCI6Ii8zMzYwODA4My8zMzcyOTk0MTktMTQ2MGZkMDQtYzVhYS00NDk3LTljMTctYjQwM2M0ZGM1MmYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA5VDE0MjUyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyNWMxNTYwNTEwYmY3ODM1NDcwMGVjYmVhMjBmMWJjZDk0OWI5MjMwMzRmMjcxMDk5NTFhODYwYzhlMGEyYmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sTCHt9ayy0c0FyE_mh7qqJvhyUPquOPUKjVOXU9K4no)
> 
> I tried with and without quotes, but neither way works.

Now it is working in beta.44. Thanks to the developer. [@windingwind](https://github.com/windingwind) According to my testing, quotation marks are not required.

**Reply by DWCY3574** [reply]
> > 谢谢提醒，我更新了 readme：[https://github.com/windingwind/zotero-actions-tags](https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#action-settings) ?tab=readme-ov-file#action-settings，并发布了支持触发多个操作的新版本。
> 
> 别客气！我尝试添加两个操作，但都不起作用（什么都没发生）：
> 
> ![图像](https://private-user-images.githubusercontent.com/33608083/337299419-1460fd04-c5aa-4497-9c17-b403c4dc52f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzk1MjAsIm5iZiI6MTcyNTAzOTIyMCwicGF0aCI6Ii8zMzYwODA4My8zMzcyOTk0MTktMTQ2MGZkMDQtYzVhYS00NDk3LTljMTctYjQwM2M0ZGM1MmYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODMwVDE3MzM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYTNkNzVkNjQyMTY1YTVjYzZiNGNjODIxZjI0YzQxM2NkMjA5OWNmY2Y1MGQzNmRkYzRlYTE3MmVhYTNkOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VlLGnzyi4EgCmyDTH159T2uvWSTGPkh5njJ6gXK-7Bc)
> 
> 我尝试过使用和不使用引号，但都不起作用。

![2024-08-31_01-44-07-0](https://github.com/user-attachments/assets/61c8c21c-0346-4df3-9af3-f3c60742176f)
不用引号，可以运行

**Comment by puppytt** [comment]
这个动作好像有一点bug，当我点击文件夹，看整个文件夹的时候可以实现切换，但是当点击单个条目的时候，他就会无法实现，会出现隐藏又立马出现的bug，希望大佬能改进下

**Comment by XiXiKong** [comment]
请问把脚本加载到actions tages中，具体该如何使用

**Comment by Amo906** [comment]
Not working in the new window

**Comment by ReaderGuy42** [comment]
Would love to use this, but it's not working for me either. Are there any updates on this? Thanks :)

**Comment by rdiaz02** [comment]
Thank you. These were perfect for me with Zotero 8.0.3, and both for the items and when in the Reader. 

However, if the PDF is shown in its own window (not as a tab), the left pane toggle does not work. In fact, the scripts (that I have bound to shortcuts) affect the main Zotero window, not any of the independent PDF windows. I think this is expected?
